source/core/hash.h

```diff
@@ -7,6 +7,8 @@
 
 namespace Slang
 {
+    typedef int HashCode;
+
 	inline int GetHashCode(double key)
```

Comment:
Shouldn't these functions return 'HashCode' type? Think it's a very good idea to have a specific hash code type. That if you wanted some extra type protection around 'HashCode' you could use the enum class idiom - as in `enum class HashCode: int;` Although that might be more pain than it's worth.

Reply:
Would be a nice cleanup. I wanted to keep my diffs minimal for now while leaving the door open to subsequent improvement.